src/js/helpers/searchIndexBuilder.js

```diff
@@ -1,123 +1,287 @@
+      const {janisInstances} = node;
+      if (janisInstances.length) {
+        doc.topics = []
+        janisInstances.forEach(instance => {
+          if (instance.parent && instance.parent.title) {
+            doc.topics.push(instance.parent.title)
+          }
+        })
+      }
+    }
+
+    if (withElasticsearch) {
+      /**
+        If we're sending the searchIndex to elasticsearch, then you must add an "index" operation object
+        before the actual document that is run by that operation.
+        That's how the API works, I'm sorry.
+
+        searchIndex with elastic search looks like this:
+        [{index}, {doc}, {index}, {doc}, ...]
+
+        searchIndex WITHOUT elasticsearch looks like this:
+        [{doc}, {doc}, {doc}, {doc}, ...]
+      **/
+      searchIndex.push({
+        "index": {
+          "_index": indexName,
+          "_id": node.pageId,
+        }
+      })
     }
+    searchIndex.push(doc)
   })
-  return instanceIndex
+
+  if (withElasticsearch) {
+    await sendIndexToElasticSeach(searchIndex)
+  } else {
+    // If we're not sending the index to elasticsearch, then we're going to return it
+    // so that it can be embedded directly into the search page.
+    return searchIndex
+  }
 }
 
+/**
+  Sends searchIndex to elasticSearch server.
+  @param {Object[]} searchIndex - All of the operations + documents to add to elasticsearch index.
+**/
+const sendIndexToElasticSeach = async (searchIndex) => {
+  // Hardcoding client address for now.
+  const elasticClient = new Client({
+    node: 'http://localhost:9200'
+  })
 
+  // Taken from https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/bulk_examples.html
+  try {
+    const { body: bulkResponse } = await elasticClient.bulk({
+      refresh: true,
+      body: searchIndex,
+     })
+    if (bulkResponse.errors) {
+      const erroredDocuments = []
+      // The items array has the same order of the dataset we just indexed.
+      // The presence of the `error` key indicates that the operation
+      // that we did for the document has failed.
+      bulkResponse.items.forEach((action, i) => {
+        const operation = Object.keys(action)[0]
+        if (action[operation].error) {
+          erroredDocuments.push({
+            // If the status is 429 it means that you can retry the document,
+            // otherwise it's very likely a mapping error, and you should
+            // fix the document before to try it again.
+            status: action[operation].status,
+            error: action[operation].error,
+            operation: body[i * 2],
+            document: body[i * 2 + 1]
+          })
+        }
+      })
+      console.log(`There were errors building index ${indexName}`)
+      console.log(erroredDocuments)
+    } else {
+      console.log(`Successfully built index ${indexName}`)
+    }
+  } catch (err) {
+    console.log(`There were catastophic errors building index ${indexName}`)
```

Comment:
catastrophe!!! really gets the point across

Reply:
There are some errors that elasticsearch's API is built to handle. If there are errors outside of that, then I called them catastrophic, and the rest of the build should fail.